This patch removes all need for the ThirdParty files of OpenFOAM:
we use EB dependencies for everything. It adjusts the paths, variables, etc
We also let the install dir, compiler, etc be set by EB.
Aligned hunks by Leon Kos and updated by Simon Branford (University of Birmingham).
Based on patch for OpenFOAM 5.0 and 4.1 by Kenneth Hoste (HPC-UGent) and Ward Poelmans <wpoely86@gmail.com>
diff --git a/applications/utilities/postProcessing/graphics/PVReaders/Allwmake b/applications/utilities/postProcessing/graphics/PVReaders/Allwmake
index 1deca76ee0..bb09628dac 100755
--- a/applications/utilities/postProcessing/graphics/PVReaders/Allwmake
+++ b/applications/utilities/postProcessing/graphics/PVReaders/Allwmake
@@ -11,8 +11,8 @@ then
 fi
 
 # Ensure CMake gets the correct C/C++ compilers
-[ -n "$WM_CC" ] && export CC="$WM_CC"
-[ -n "$WM_CXX" ] && export CXX="$WM_CXX"
+#[ -n "$WM_CC" ] && export CC="$WM_CC"
+#[ -n "$WM_CXX" ] && export CXX="$WM_CXX"
 
 wmake $targetType vtkPVblockMesh
 wmake $targetType vtkPVFoam
diff --git a/applications/utilities/postProcessing/graphics/PVReaders/CMakeLists.txt b/applications/utilities/postProcessing/graphics/PVReaders/CMakeLists.txt
index e199b3f331..5328d4880c 100644
--- a/applications/utilities/postProcessing/graphics/PVReaders/CMakeLists.txt
+++ b/applications/utilities/postProcessing/graphics/PVReaders/CMakeLists.txt
@@ -2,6 +2,7 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.5.1)
 
 PROJECT(PVReaders)
 
+FIND_PACKAGE(MPI REQUIRED)
 FIND_PACKAGE(ParaView REQUIRED)
 
 INCLUDE(GNUInstallDirs)
diff --git a/etc/bashrc b/etc/bashrc
index e77c0a8626..77467c06e2 100644
--- a/etc/bashrc
+++ b/etc/bashrc
@@ -44,12 +44,13 @@ export WM_PROJECT_VERSION=13
 #
 [ "$BASH" ] && bashrcFile=${BASH_SOURCE}
 [ "$ZSH_NAME" ] && bashrcFile=$0
-if [ -n "$bashrcFile" ]
-then
-    export FOAM_INST_DIR=$(cd $(dirname $bashrcFile)/../.. && pwd -P)
-else
-    export FOAM_INST_DIR=$HOME/$WM_PROJECT
-fi
+#if [ -n "$bashrcFile" ]
+#then
+#    export FOAM_INST_DIR=$(cd $(dirname $bashrcFile)/../.. && pwd -P)
+#else
+#    export FOAM_INST_DIR=$HOME/$WM_PROJECT
+#fi
+## For Easybuild: set by the module
 unset bashrcFile
 #
 # Please set to the appropriate path if the above default is not correct. E.g.,
diff --git a/etc/config.sh/gperftools b/etc/config.sh/gperftools
index f9cdb85334..1232b118c3 100644
--- a/etc/config.sh/gperftools
+++ b/etc/config.sh/gperftools
@@ -29,13 +29,15 @@
 #
 #------------------------------------------------------------------------------
 
-version=svn
-gperftools_install=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER
-
-GPERFTOOLS_VERSION=gperftools-$version
-GPERFTOOLS_ARCH_PATH=$gperftools_install/$GPERFTOOLS_VERSION
-
-export PATH=$GPERFTOOLS_ARCH_PATH/bin:$PATH
-export LD_LIBRARY_PATH=$GPERFTOOLS_ARCH_PATH/lib:$LD_LIBRARY_PATH
+#version=svn
+#gperftools_install=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER
+#
+#GPERFTOOLS_VERSION=gperftools-$version
+#GPERFTOOLS_ARCH_PATH=$gperftools_install/$GPERFTOOLS_VERSION
+#
+#export PATH=$GPERFTOOLS_ARCH_PATH/bin:$PATH
+#export LD_LIBRARY_PATH=$GPERFTOOLS_ARCH_PATH/lib:$LD_LIBRARY_PATH
+GPERFTOOLS_VERSION=gperftools-$EBVERSIONGPERFTOOLS
+GPERFTOOLS_ARCH_PATH=$EBROOTGPERFTOOLS
 
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/metis b/etc/config.sh/metis
index a40707ac55..b8a7ec71c5 100644
--- a/etc/config.sh/metis
+++ b/etc/config.sh/metis
@@ -38,31 +38,33 @@
 . $WM_PROJECT_DIR/etc/config.sh/functions
 
 # Find the path to the metis installation
-case "$METIS_TYPE" in
-none)
-    ;;
-system)
-    export METIS_VERSION=system
-    export METIS_ARCH_PATH=/usr
-    ;;
-OpenFOAM | ThirdParty)
-    # Look for the source directory
-    if [ -z "$METIS_VERSION" ]
-    then
-        metisSrcDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/metis-*)
-    else
-        metisSrcDir=$WM_THIRD_PARTY_DIR/metis-$METIS_VERSION
-    fi
-    # Set the version and the installation path
-    if [ -d "$metisSrcDir" ]
-    then
-        export METIS_VERSION=${metisSrcDir##*metis-}
-        export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/metis-$METIS_VERSION
-    fi
-    # Clean up
-    unset metisSrcDir
-    ;;
-esac
+#case "$METIS_TYPE" in
+#none)
+#    ;;
+#system)
+#    export METIS_VERSION=system
+#    export METIS_ARCH_PATH=/usr
+#    ;;
+#OpenFOAM | ThirdParty)
+#    # Look for the source directory
+#    if [ -z "$METIS_VERSION" ]
+#    then
+#        metisSrcDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/metis-*)
+#    else
+#        metisSrcDir=$WM_THIRD_PARTY_DIR/metis-$METIS_VERSION
+#    fi
+#    # Set the version and the installation path
+#    if [ -d "$metisSrcDir" ]
+#    then
+#        export METIS_VERSION=${metisSrcDir##*metis-}
+#        export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/metis-$METIS_VERSION
+#    fi
+#    # Clean up
+#    unset metisSrcDir
+#    ;;
+#esac
+export METIS_VERSION=metis-$EBVERSIONMETIS
+export METIS_ARCH_PATH=$EBROOTMETIS
 
 # Unload functions
 . $WM_PROJECT_DIR/etc/config.sh/functions
diff --git a/etc/config.sh/mpi b/etc/config.sh/mpi
index f71d7cfd2d..3acdca2904 100644
--- a/etc/config.sh/mpi
+++ b/etc/config.sh/mpi
@@ -299,6 +299,10 @@ INTELMPI)
     _foamAddPath    $MPI_ARCH_PATH/bin64
     _foamAddLib     $MPI_ARCH_PATH/lib/release
     ;;
+
+EASYBUILDMPI)
+    export FOAM_MPI=mpi
+    ;;
 *)
     export FOAM_MPI=dummy
     ;;
diff --git a/etc/config.sh/paraview b/etc/config.sh/paraview
index 5cac9f8b4e..2ff12de312 100644
--- a/etc/config.sh/paraview
+++ b/etc/config.sh/paraview
@@ -47,108 +47,115 @@ fi
 unset cmakeDir
 
 # Set up the paraview environment
-case "$ParaView_TYPE" in
-none)
-    ;;
-
-system)
-
-    # Look for a paraview installation
-    if pvserverExe=$(which pvserver 2> /dev/null)
-    then
-        paraviewBinDir=$(dirname $pvserverExe)
-        paraviewBinDir=$(cd $paraviewBinDir && pwd -P)
-    fi
-
-    # Set the environment
-    if [ -d "$paraviewBinDir" ]
-    then
-        export ParaView_DIR=$(dirname $paraviewBinDir)
-        export ParaView_LIB_DIR=$(unset LD_LIBRARY_PATH && \
-                                  ldd $paraviewBinDir/paraview | \
-                                  grep -o "/.*/libpqCore-pv.*.so" | \
-                                  xargs dirname)
-        export ParaView_VERSION=$(unset LD_LIBRARY_PATH && \
-                                  pvserver --version 2> /dev/null | \
-                                  awk '{print $NF}')
-        export ParaView_MAJOR=${ParaView_VERSION%.*}
-        export ParaView_INCLUDE_DIR=$ParaView_DIR/include/paraview-$ParaView_MAJOR
-        export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
-
-        # Add to the library path
-        export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
-    fi
-
-    unset pvserverExe paraviewBinDir
-    ;;
-
-paraviewopenfoam)
-
-    # Look for a paraview installation. The version should be set externally.
-    if [ -n "$ParaView_VERSION" ]
-    then
-        export ParaView_MAJOR=${ParaView_VERSION%.*}
-        paraviewDir=/opt/paraviewopenfoam$(echo "$ParaView_MAJOR" | tr -d '.')
-    fi
-
-    # Set the environment
-    if [ -d "$paraviewDir" ]
-    then
-        export ParaView_DIR=$paraviewDir
-        export ParaView_LIB_DIR=$(echo "$ParaView_DIR"/lib* | tr ' ' ':')
-        export ParaView_INCLUDE_DIR=$ParaView_DIR/include/paraview-$ParaView_MAJOR
-        export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
-
-        # Add to the path and the library path
-        export PATH=$ParaView_DIR/bin:$PATH
-        export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
-    fi
-
-    unset paraviewDir
-    ;;
-
-OpenFOAM | ThirdParty)
-
-    # Look for a paraview installation
-    if [ -z "$ParaView_VERSION" ]
-    then
-        paraviewDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/platforms/$WM_ARCH$WM_COMPILER/ParaView-*)
-    else
-        paraviewDir=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/ParaView-$ParaView_VERSION
-    fi
-
-    # Set the environment
-    if [ -d "$paraviewDir" ]
-    then
-        export ParaView_DIR=$paraviewDir
-        export ParaView_LIB_DIR=$(echo "$ParaView_DIR"/lib* | tr ' ' ':')
-        export ParaView_VERSION=${paraviewDir##*ParaView-}
-        export ParaView_MAJOR=${ParaView_VERSION%.*}
-        export ParaView_INCLUDE_DIR=$paraviewDir/include/paraview-$ParaView_MAJOR
-        export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
-
-        # Add to the path and the library path
-        export PATH=$ParaView_DIR/bin:$PATH
-        export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
-
-        # Add in python libraries if required
-        paraviewPython=$ParaView_DIR/Utilities/VTKPythonWrapping
-        if [ -r "$paraviewPython" ]
-        then
-            if [ "$PYTHONPATH" ]
-            then
-                export PYTHONPATH=$PYTHONPATH:$paraviewPython:$ParaView_LIB_DIR
-            else
-                export PYTHONPATH=$paraviewPython:$ParaView_LIB_DIR
-            fi
-        fi
-        unset paraviewPython
-    fi
-
-    unset paraviewDir
-    ;;
-
-esac
+#case "$ParaView_TYPE" in
+#none)
+#    ;;
+#
+#system)
+#
+#    # Look for a paraview installation
+#    if pvserverExe=$(which pvserver 2> /dev/null)
+#    then
+#        paraviewBinDir=$(dirname $pvserverExe)
+#        paraviewBinDir=$(cd $paraviewBinDir && pwd -P)
+#    fi
+#
+#    # Set the environment
+#    if [ -d "$paraviewBinDir" ]
+#    then
+#        export ParaView_DIR=$(dirname $paraviewBinDir)
+#        export ParaView_LIB_DIR=$(unset LD_LIBRARY_PATH && \
+#                                  ldd $paraviewBinDir/paraview | \
+#                                  grep -o "/.*/libpqCore-pv.*.so" | \
+#                                  xargs dirname)
+#        export ParaView_VERSION=$(unset LD_LIBRARY_PATH && \
+#                                  pvserver --version 2> /dev/null | \
+#                                  awk '{print $NF}')
+#        export ParaView_MAJOR=${ParaView_VERSION%.*}
+#        export ParaView_INCLUDE_DIR=$ParaView_DIR/include/paraview-$ParaView_MAJOR
+#        export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
+#
+#        # Add to the library path
+#        export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
+#    fi
+#
+#    unset pvserverExe paraviewBinDir
+#    ;;
+#
+#paraviewopenfoam)
+#
+#    # Look for a paraview installation. The version should be set externally.
+#    if [ -n "$ParaView_VERSION" ]
+#    then
+#        export ParaView_MAJOR=${ParaView_VERSION%.*}
+#        paraviewDir=/opt/paraviewopenfoam$(echo "$ParaView_MAJOR" | tr -d '.')
+#    fi
+#
+#    # Set the environment
+#    if [ -d "$paraviewDir" ]
+#    then
+#        export ParaView_DIR=$paraviewDir
+#        export ParaView_LIB_DIR=$(echo "$ParaView_DIR"/lib* | tr ' ' ':')
+#        export ParaView_INCLUDE_DIR=$ParaView_DIR/include/paraview-$ParaView_MAJOR
+#        export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
+#
+#        # Add to the path and the library path
+#        export PATH=$ParaView_DIR/bin:$PATH
+#        export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
+#    fi
+#
+#    unset paraviewDir
+#    ;;
+#
+#OpenFOAM | ThirdParty)
+#
+#    # Look for a paraview installation
+#    if [ -z "$ParaView_VERSION" ]
+#    then
+#        paraviewDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/platforms/$WM_ARCH$WM_COMPILER/ParaView-*)
+#    else
+#        paraviewDir=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/ParaView-$ParaView_VERSION
+#    fi
+#
+#    # Set the environment
+#    if [ -d "$paraviewDir" ]
+#    then
+#        export ParaView_DIR=$paraviewDir
+#        export ParaView_LIB_DIR=$(echo "$ParaView_DIR"/lib* | tr ' ' ':')
+#        export ParaView_VERSION=${paraviewDir##*ParaView-}
+#        export ParaView_MAJOR=${ParaView_VERSION%.*}
+#        export ParaView_INCLUDE_DIR=$paraviewDir/include/paraview-$ParaView_MAJOR
+#        export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
+#
+#        # Add to the path and the library path
+#        export PATH=$ParaView_DIR/bin:$PATH
+#        export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
+#
+#        # Add in python libraries if required
+#        paraviewPython=$ParaView_DIR/Utilities/VTKPythonWrapping
+#        if [ -r "$paraviewPython" ]
+#        then
+#            if [ "$PYTHONPATH" ]
+#            then
+#                export PYTHONPATH=$PYTHONPATH:$paraviewPython:$ParaView_LIB_DIR
+#            else
+#                export PYTHONPATH=$paraviewPython:$ParaView_LIB_DIR
+#            fi
+#        fi
+#        unset paraviewPython
+#    fi
+#
+#    unset paraviewDir
+#    ;;
+set (PARAVIEW_QT_MAJOR_VERSION,6)
+#esac
+export ParaView_VERSION=$EBVERSIONPARAVIEW
+export ParaView_MAJOR=${ParaView_VERSION%.*}
+export ParaView_DIR=$EBROOTPARAVIEW
+export ParaView_LIB_DIR=$ParaView_DIR/lib
+export ParaView_INCLUDE_DIR=$ParaView_DIR/include/paraview-$ParaView_MAJOR
+export PV_PLUGIN_PATH=$FOAM_LIBBIN/paraview-$ParaView_MAJOR
+export LD_LIBRARY_PATH=$ParaView_LIB_DIR:$PV_PLUGIN_PATH:$LD_LIBRARY_PATH
 
 # Report
 if [ "$FOAM_VERBOSE" ] && [ "$PS1" ] && [ -d "$ParaView_DIR" ]
diff --git a/etc/config.sh/scotch b/etc/config.sh/scotch
index 7693417ed4..10d74af470 100644
--- a/etc/config.sh/scotch
+++ b/etc/config.sh/scotch
@@ -38,31 +38,33 @@
 . $WM_PROJECT_DIR/etc/config.sh/functions
 
 # Find the path to the scotch installation
-case "$SCOTCH_TYPE" in
-none)
-    ;;
-system)
-    export SCOTCH_VERSION=system
-    export SCOTCH_ARCH_PATH=/usr
-    ;;
-OpenFOAM | ThirdParty)
-    # Look for the source directory
-    if [ -z "$SCOTCH_VERSION" ]
-    then
-        scotchSrcDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/scotch_*)
-    else
-        scotchSrcDir=$WM_THIRD_PARTY_DIR/scotch_$SCOTCH_VERSION
-    fi
-    # Set the version and the installation path
-    if [ -d "$scotchSrcDir" ]
-    then
-        export SCOTCH_VERSION=${scotchSrcDir##*scotch_}
-        export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/scotch_$SCOTCH_VERSION
-    fi
-    # Clean up
-    unset scotchSrcDir
-    ;;
-esac
+#case "$SCOTCH_TYPE" in
+#none)
+#    ;;
+#system)
+#    export SCOTCH_VERSION=system
+#    export SCOTCH_ARCH_PATH=/usr
+#    ;;
+#OpenFOAM | ThirdParty)
+#    # Look for the source directory
+#    if [ -z "$SCOTCH_VERSION" ]
+#    then
+#        scotchSrcDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/scotch_*)
+#    else
+#        scotchSrcDir=$WM_THIRD_PARTY_DIR/scotch_$SCOTCH_VERSION
+#    fi
+#    # Set the version and the installation path
+#    if [ -d "$scotchSrcDir" ]
+#    then
+#        export SCOTCH_VERSION=${scotchSrcDir##*scotch_}
+#        export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/scotch_$SCOTCH_VERSION
+#    fi
+#    # Clean up
+#    unset scotchSrcDir
+#    ;;
+#esac
+export SCOTCH_VERSION=scotch_$EBVERSIONSCOTCH
+export SCOTCH_ARCH_PATH=$EBROOTSCOTCH
 
 # Unload functions
 . $WM_PROJECT_DIR/etc/config.sh/functions
diff --git a/etc/config.sh/settings b/etc/config.sh/settings
index 0a69a70190..8b6962a17d 100644
--- a/etc/config.sh/settings
+++ b/etc/config.sh/settings
@@ -63,11 +63,11 @@ Linux)
         64)
             WM_ARCH=linux64
             export WM_COMPILER_LIB_ARCH=64
-            export WM_CC='gcc'
-            export WM_CXX='g++'
-            export WM_CFLAGS='-m64 -fPIC'
-            export WM_CXXFLAGS='-m64 -fPIC -std=c++0x'
-            export WM_LDFLAGS='-m64'
+            export WM_CC=$CC
+            export WM_CXX=$CXX
+            export WM_CFLAGS=$CFLAGS
+            export WM_CXXFLAGS=$CXXFLAGS
+            export WM_LDFLAGS=$LDFLAGS
             ;;
 
         *)
diff --git a/etc/config.sh/zoltan b/etc/config.sh/zoltan
index 6103cb5762..59bc7fa526 100644
--- a/etc/config.sh/zoltan
+++ b/etc/config.sh/zoltan
@@ -38,31 +38,33 @@
 . $WM_PROJECT_DIR/etc/config.sh/functions
 
 # Find the path to the zoltan installation
-case "$ZOLTAN_TYPE" in
-none)
-    ;;
-system)
-    export ZOLTAN_VERSION=system
-    export ZOLTAN_ARCH_PATH=/usr
-    ;;
-OpenFOAM | ThirdParty)
-    # Look for the source directory
-    if [ -z "$ZOLTAN_VERSION" ]
-    then
-        zoltanSrcDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/Zoltan-*)
-    else
-        zoltanSrcDir=$WM_THIRD_PARTY_DIR/Zoltan-$ZOLTAN_VERSION
-    fi
-    # Set the version and the installation path
-    if [ -d "$zoltanSrcDir" ]
-    then
-        export ZOLTAN_VERSION=${zoltanSrcDir##*Zoltan-}
-        export ZOLTAN_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/Zoltan-$ZOLTAN_VERSION
-    fi
-    # Clean up
-    unset zoltanSrcDir
-    ;;
-esac
+#case "$ZOLTAN_TYPE" in
+#none)
+#    ;;
+#system)
+#    export ZOLTAN_VERSION=system
+#    export ZOLTAN_ARCH_PATH=/usr
+#    ;;
+#OpenFOAM | ThirdParty)
+#    # Look for the source directory
+#    if [ -z "$ZOLTAN_VERSION" ]
+#    then
+#        zoltanSrcDir=$(_foamMostRecentDir "$WM_THIRD_PARTY_DIR"/Zoltan-*)
+#    else
+#        zoltanSrcDir=$WM_THIRD_PARTY_DIR/Zoltan-$ZOLTAN_VERSION
+#    fi
+#    # Set the version and the installation path
+#    if [ -d "$zoltanSrcDir" ]
+#    then
+#        export ZOLTAN_VERSION=${zoltanSrcDir##*Zoltan-}
+#        export ZOLTAN_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/Zoltan-$ZOLTAN_VERSION
+#    fi
+#    # Clean up
+#    unset zoltanSrcDir
+#    ;;
+#esac
+export ZOLTAN_VERSION=Zoltan-$EBVERSIONZOLTAN
+export ZOLTAN_ARCH_PATH=$EBROOTZOLTAN
 
 # Unload functions
 . $WM_PROJECT_DIR/etc/config.sh/functions
diff --git a/src/parallel/decompose/ptscotch/Make/options b/src/parallel/decompose/ptscotch/Make/options
index 59ab239980..03d506fe60 100644
--- a/src/parallel/decompose/ptscotch/Make/options
+++ b/src/parallel/decompose/ptscotch/Make/options
@@ -7,15 +7,12 @@ EXE_INC = \
     $(PFLAGS) $(PINC) \
     -I$(FOAM_SRC)/Pstream/mpi/lnInclude \
     -I$(SCOTCH_ARCH_PATH)/include/$(FOAM_MPI) \
-    -I$(SCOTCH_ARCH_PATH)/include \
-    -I$(or $(PTSCOTCH_INCLUDE_DIR),/usr/include/scotch) \
     -I../decompositionMethods/lnInclude
 
 LIB_LIBS = \
     -L$(SCOTCH_ARCH_PATH)/lib \
-    $(if $(PTSCOTCH_LIB_DIR),-L$(PTSCOTCH_LIB_DIR)) \
     -L$(FOAM_EXT_LIBBIN)/$(FOAM_MPI) \
     -lptscotch \
     -lptscotcherrexit \
-    -lscotch \
+    -lscotch ${LINK_FLAGS} \
     -lrt
diff --git a/src/parallel/decompose/ptscotch/ptscotch.C b/src/parallel/decompose/ptscotch/ptscotch.C
index 7f6738c9d3..2f05daa926 100644
--- a/src/parallel/decompose/ptscotch/ptscotch.C
+++ b/src/parallel/decompose/ptscotch/ptscotch.C
@@ -30,12 +30,13 @@ License
 #include "globalIndex.H"
 #include "SubField.H"
 #include "PstreamGlobals.H"
+    
+#include <mpi.h>
+#include "ptscotch.h"
 
 extern "C"
 {
     #include <stdio.h>
-    #include <mpi.h>
-    #include "ptscotch.h"
 }
 
 
diff --git a/src/parallel/decompose/scotch/Make/options b/src/parallel/decompose/scotch/Make/options
index 79b724e4e7..9c5377cd16 100644
--- a/src/parallel/decompose/scotch/Make/options
+++ b/src/parallel/decompose/scotch/Make/options
@@ -6,7 +6,6 @@
 EXE_INC = \
     $(PFLAGS) $(PINC) \
     -I$(SCOTCH_ARCH_PATH)/include \
-    -I$(or $(SCOTCH_INCLUDE_DIR),/usr/include/scotch) \
     -I../decompositionMethods/lnInclude
 
 LIB_LIBS = \
